### PR TITLE
Enable build filter

### PR DESCRIPTION
**Pull request by cormacpayne**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by MiYanni** [comment]
@cormacpayne Is this ready enough to be reviewed?

File: .azure-pipelines/util/analyze-steps.yml [hunk 1]
```diff
@@ -15,14 +15,14 @@ steps:
   inputs:
     command: custom
     custom: msbuild
-    arguments: 'build.proj /t:GenerateHelp /p:Configuration=${{ parameters.configuration }}'
+    arguments: 'build.proj /t:GenerateHelp /p:Configuration=${{ parameters.configuration }};PullRequestNumber=$(System.PullRequest.PullRequestNumber)'
```
Comment (MiYanni) [inline .azure-pipelines/util/analyze-steps.yml]: Could you verify this works on non-PullRequest builds? Put these changes to a branch in Azure, and then queue the build in ADO manually, targeting that branch.

File: tools/CreateFilterMappings.ps1 [hunk 2]
```diff
@@ -107,6 +88,21 @@ function Create-Key
     return $Key
 }
 
+function Create-ProjectToFullPathMappings
+{
+    $Mappings = [ordered]@{}
+    foreach ($ServiceFolder in $Script:ServiceFolders)
+    {
+        $CsprojFiles = Get-ChildItem -Path $ServiceFolder -Filter "*.csproj" -Recurse | where { $_.FullName -notlike "*Stack*" }
```
Comment (MiYanni) [inline tools/CreateFilterMappings.ps1]: Likely do not need the Stack filter anymore.

File: tools/CreateFilterMappings.ps1 [hunk 3]
```diff
@@ -128,20 +124,18 @@ function Create-SolutionToProjectMappings
     $Mappings = [ordered]@{}
     foreach ($ServiceFolder in $Script:ServiceFolders)
     {
-        $SolutionFiles = Get-ChildItem -Path $ServiceFolder.FullName -Filter "*.sln" | where { $_.FullName -notlike "*Stack*" -and $_.FullName -notlike "*Netcore*" }
+        $SolutionFiles = Get-ChildItem -Path $ServiceFolder.FullName -Filter "*.sln" | Where-Object { $_.FullName -notlike "*Stack*" }
```
Comment (MiYanni) [inline tools/CreateFilterMappings.ps1]: Likely do not need the Stack filter anymore.

File: tools/CreateFilterMappings.ps1 [hunk 4]
```diff
@@ -221,214 +215,81 @@ Creates the ModuleMappings.json file used during the build to filter StaticAnaly
...
-    $FilteredServiceFolders = $Script:ServiceFolders
-    foreach ($ServiceFolder in $FilteredServiceFolders)
+    $CustomMappings = @{}
+    $Script:ModuleMappings = Initialize-Mappings -PathsToIgnore $PathsToIgnore -CustomMappings $CustomMappings
+    foreach ($ServiceFolder in $Script:ServiceFolders)
     {
         $Key = "src/$($ServiceFolder.Name)"
-        $ModuleManifestFiles = Get-ChildItem -Path $ServiceFolder.FullName -Filter "*.psd1" -Recurse | where { $_.FullName -notlike "*Stack*" -and $_.Name -like "*Azure*" -and $_.FullName -notlike "*Test*" }
-        if ($ModuleManifestFiles -ne $null)
+        $ModuleManifestFiles = Get-ChildItem -Path $ServiceFolder.FullName -Filter "*.psd1" -Recurse | Where-Object { $_.FullName -notlike "*Stack*" -and `
```
Comment (MiYanni) [inline tools/CreateFilterMappings.ps1]: Likely do not need the Stack filter anymore.

File: .azure-pipelines/util/analyze-steps.yml (hunk 1 above)
Comment (cormacpayne) [reply .azure-pipelines/util/analyze-steps.yml]: Manual build here: https://dev.azure.com/azure-sdk/public/_build/results?buildId=38001

**Pull request merged**